### PR TITLE
Enhancement/metadata hash

### PR DESCRIPTION
**Pull request by Sital999**
- Auto-generate hash for correct URL of metadata
- Handle error on incorrect URL of metadata

### CODE REVIEW

File: api/backend/app/controllers/internal/metadata_router.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        super().__init__()
+        self.metadata_api = APISettings().METADATA_API
+
+    @get("/metadata")
+    async def fetch_metadata(self, metadata_url: str):
+        async with aiohttp.ClientSession() as session:
+            async with session.get(
+                f"{self.metadata_api}/metadata?url={metadata_url}&hash=1111111111111111111111111111111111111111111111111111111111111112"
+            ) as resp:
+                response = await resp.json()
```
Comment (mesudip) [inline api/backend/app/controllers/internal/metadata_router.py:20]: Return the error message returned in the response instead.

File: frontend/src/app/api/metaData.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { baseAPIurl } from './config';
+
+export async function fetchMetadataHash(url: string) {
+    const res = await fetch(`${baseAPIurl}/metadata?metadata_url=${url}`);
```
Comment (mesudip) [inline frontend/src/app/api/metaData.ts:4]: use encodeURIComponent function just in case.

**Review by mesudip — CHANGES_REQUESTED** [review]

File: api/backend/app/controllers/internal/metadata_router.py (hunk 1 above)
Comment (Sital999) [reply api/backend/app/controllers/internal/metadata_router.py:20]: Done

File: frontend/src/app/api/metaData.ts (hunk 2 above)
Comment (Sital999) [reply frontend/src/app/api/metaData.ts]: Done

**Review by mesudip — APPROVED** [review]

**Pull request merged**